### PR TITLE
MEPTS-1493 | NEW MQ Category 19 

### PR DESCRIPTION
**Pull request by TinoMuianga**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java [hunk 1]
```diff
@@ -2741,4 +2741,164 @@ public static String getPatientsWithRestartedStateOfStayAndCd4RequestQuery() {
...
+        + "          AND e.voided = 0 "
+        + "          AND o.voided = 0 "
+        + "          AND e.location_id = :location "
+        + "          AND e.encounter_datetime >= :startDate "
+        + "          AND e.encounter_datetime <= :endDate "
+        + "          AND e.encounter_type = ${6} "
+        + "          AND o.concept_id = ${23758} "
+        + "          AND o.value_coded = ${1065} "
+        + "        GROUP  BY p.patient_id";
+  }
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java]: Can you create one function that receives 2 params, a concept id and list of concepts as value coded a will produce your patient id and min(encounter_datetime)

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
Please have a look @TinoMuianga

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java (hunk 1 above)
Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java]: Done

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**